### PR TITLE
Take over Google's wdl_runner

### PR DESCRIPTION
**Pull request by vdauwera**
This copies the code from https://github.com/googlegenomics/pipelines-api-examples/ and renames "wdl_runner" to "cromwell-on-google" to be more specific.

### CODE REVIEW

**Comment by vdauwera** [comment]
@mbookman FYI

**Comment by mbookman** [comment]
The most recent commit removed `runners/cromwell_on_google/workflows/wdl_pipeline.yaml`.
That is needed to run workflows from gcloud.
Can that file be restored?

**Comment by vdauwera** [comment]
@mbookman I updated this, afaict everything works (with Cromwell 29). Would you care to review?

**Comment by vdauwera** [comment]
@mbookman Updated the PR based on our discussion earlier today

File: runners/cromwell_on_google/wdl_runner/README.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+### Code summary
+
+The code in the wdl_runner Docker image includes:
+
+* [OpenJDK 8](http://openjdk.java.net/projects/jdk8/) runtime engine (JRE)
+* [Python 2.7](https://www.python.org/download/releases/2.7/) interpreter
+* [Cromwell release 29](https://github.com/broadinstitute/cromwell/releases/tag/29)
+* [Python and shell scripts from this repository](./wdl_runner)
+
+Take a look at the [Dockerfile](./wdl_runner/Dockerfile) for full details.
```
Comment (mbookman) [inline runners/cromwell_on_google/wdl_runner/README.md]: I think this link should just be ./Dockerfile

File: runners/cromwell_on_google/wdl_runner/README.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+the Broad Institute.
+
+### Code summary
+
+The code in the wdl_runner Docker image includes:
+
+* [OpenJDK 8](http://openjdk.java.net/projects/jdk8/) runtime engine (JRE)
+* [Python 2.7](https://www.python.org/download/releases/2.7/) interpreter
+* [Cromwell release 29](https://github.com/broadinstitute/cromwell/releases/tag/29)
+* [Python and shell scripts from this repository](./wdl_runner)
```
Comment (mbookman) [inline runners/cromwell_on_google/wdl_runner/README.md]: This path should just be "."

File: runners/cromwell_on_google/wdl_runner/README.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+The output will be an operation ID for the Pipeline.
+
+## (3) Monitor the pipeline operation
+
+This github repo includes a shell script,
+[./monitoring_tools/monitor_wdl_pipelines.sh](./monitoring_tools/monitor_wdl_pipelines.sh),
+for monitoring the status of a pipeline launched using ``wdl_pipeline.yaml``.
+
+```
+$ ./tools/monitor_wdl_pipeline.sh YOUR-NEW-OPERATION-ID
```
Comment (mbookman) [inline runners/cromwell_on_google/wdl_runner/README.md]: s/tools/monitoring_tools

File: runners/cromwell_on_google/wdl_runner/test-wdl/basic.papi.us.options.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "default_runtime_attributes": {
+    "zones": "us-central1-a us-central1-b us-central1-c us-central1-f us-east1-b us-east1-c us-east1-d us-west1-a us-west1-b"
```
Comment (mbookman) [inline runners/cromwell_on_google/wdl_runner/test-wdl/basic.papi.us.options.json]: Might want to change this to a specific region, such as us-central1.
Before DRA was deprecated, using any US zone made sense as data was typically in a Standard or DRA US bucket.

With Google's encouragement now of Regional buckets as a replacement for DRA, one should be more careful about which zones they run their compute in. Running Compute in a different region than the user's bucket will generate unnecessary egress charges.

See https://cloud.google.com/compute/pricing#network

**Comment by vdauwera** [comment]
Good eye, @mbookman -- thanks for catching these. I think I addressed all your points.

File: runners/cromwell_on_google/wdl_runner/test-wdl/basic.papi.us.options.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "default_runtime_attributes": {
-    "zones": "us-central1-a us-central1-b us-central1-c us-central1-f us-east1-b us-east1-c us-east1-d us-west1-a us-west1-b" 
+    "zones": "us-central1"
```
Comment (mbookman) [inline runners/cromwell_on_google/wdl_runner/test-wdl/basic.papi.us.options.json]: Sorry my comment was misleading.
As far as I know, Cromwell doesn't support specification of a region here.
You would need to list the us-central1 zones explicitly:

    "zones": "us-central1-a us-central1-b us-central1-c us-central1-f"

Comment (vdauwera) [reply runners/cromwell_on_google/wdl_runner/test-wdl/basic.papi.us.options.json]: ah that's disappointing, I thought you were giving me a nifty trick there ;)

File: runners/cromwell_on_google/monitoring_tools/get_yaml_value.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+#
+# Utility script for extracting values from YAML.
+# This will typically be called from shell scripts, which typically
+# have fairly hacky ways of extracting values from YAML.
+#
+# An example usage would be where a shell script has YAML output from
+# a gcloud command for an genomics operation and needs to extract fields:
+#
+#  OP=$(gcloud --format=yaml alpha genomics operations describe "${OP_ID}")
+#  CTIME=$(python tools/get_yaml_value.py "${OP}" "metadata.createTime")
```
Comment (mbookman) [inline runners/cromwell_on_google/monitoring_tools/get_yaml_value.py]: Here are two more references to the "tools" directory. These can both just be removed.
There is nothing in this example that needs to imply that you are in the cromwell_on_google directory:

    CTIME=$(python get_yaml_value.py "${OP}" "metadata.createTime")
    ETIME=$(python get_yaml_value.py "${OP}" "metadata.endTime")

File: runners/cromwell_on_google/monitoring_tools/get_yaml_value.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+#
+# Note that gcloud directly supports extracting fields, so the above could also
+# be:
+#
+#  CTIME=$(gcloud alpha genomics operations describe "${OP_ID}"
+#           --format='value(metadata.createTime)')
+#  ETIME=$(gcloud alpha genomics operations describe "${OP_ID}"
+#          --format='value(metadata.endTime)')
+#
+# but then requires an API calls to get each value.
```
Comment (mbookman) [inline runners/cromwell_on_google/monitoring_tools/get_yaml_value.py]: s/calls/call

**Comment by vdauwera** [comment]
Thanks @mbookman

**Pull request merged**